### PR TITLE
Detect Scarb version from `.tool-versions` file

### LINKED ISSUE #1: Detect Scarb version from .tool-versions file
Like `setup-ruby` does with `.ruby-version`

https://github.com/ruby/setup-ruby

### PR DESCRIPTION
**Pull request by szymmis**
Resolves #1.

The action will now be capable of reading Scarb version from `.tool-versions` when it is not specified in the workflow file as `scarb-version` parameter is no longer required.

Steps of determining Scarb version in order of importance:
1. `scarb-version` parameter in action yml file
2. `.tool-version` file
3. fetching latest version from GitHub as a fallback

When `scarb-version` is not specified, action will look for `.tool-version` file in the root directory of the repository and if it fails to find the file or the scarb entry it will fallback to the latest version of Scarb.

### CODE REVIEW

File: lib/versions.js [hunk 1]
```diff
@@ -60,3 +60,15 @@ function fetchLatestTag(repo) {
     },
   );
 }
+
+function getVersionFromToolVersionsFile() {
+  const fileName = ".tool-versions";
+  if (!fs.existsSync(fileName)) return;
+
+  const toolVersions = fs.readFileSync(fileName, "utf-8").split("\n");
+  const scarbVersionInfo = toolVersions.find((tool) => tool.match(/scarb/));
```
Comment (maciektr) [inline lib/versions.js:69]: Make sure this only matches lines that start with `scarb` and ignores comments.

e.g. https://github.com/asdf-vm/actions/blob/8b8467c5522bb0aea4977082037e4f2956d67e52/src/plugins-add/index.ts#L50C23-L50C23

Comment (szymmis) [reply lib/versions.js]: Good catch, thanks!

File: lib/versions.js [hunk 2]
```diff
@@ -60,3 +60,15 @@ function fetchLatestTag(repo) {
     },
   );
 }
+
+function getVersionFromToolVersionsFile() {
+  const fileName = ".tool-versions";
+  if (!fs.existsSync(fileName)) return;
+
+  const toolVersions = fs.readFileSync(fileName, "utf-8").split("\n");
+  const scarbVersionInfo = toolVersions.find((tool) => tool.match(/^scarb/));
```
Comment (mkaput) [inline lib/versions.js:69]: nitpick, [`match` returns `null` if no matches are found](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#return_value), so this `find` call can be removed

I'm talking about this not for micro-optimization purposes; but rather I'm afraid of having two different regexes being used here

File: lib/versions.js
Comment (mkaput) [inline lib/versions.js:1]: Can we have a test for this change? I suggest adding next steps in the test job in Ci workflow.

File: lib/versions.js (hunk 2 above)
Comment (maciektr) [inline lib/versions.js:69]: Nit: please add space, so we only match `scarb ` exactly. Can be just `.startsWith`.

File: lib/versions.js [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-    throw new Error(`'scarb-input' value must not be empty`);
+    let toolVersion = getVersionFromToolVersionsFile();
+    versionInput = toolVersion ?? (await fetchLatestTag(repo));
   }
 
-  return versionInput;
+  return versionInput.startsWith("v")
+    ? versionInput.substring(1)
+    : versionInput;
 }
```
Comment (mkaput) [inline lib/versions.js:20]: I believe this doesn't work if user does not specify `scarb-version` and has no `.tool-versions` file. We still should print a meaningful error in this case, telling the user what is wrong and how to fix it (I now see that it was missing, but please add it)

**Review by mkaput — CHANGES_REQUESTED** [review]

File: lib/versions.js (hunk 3 above)
Comment (mkaput) [reply lib/versions.js:20]: Ok, @maciektr  pointed out to me that this works as expected. Ignore this comment.

File: lib/versions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mkaput) [reply lib/versions.js:69]: @maciektr told me that nobody cares about this whitespace, so let's ignore this comment :D

File: lib/versions.js (hunk 3 above)
Comment (maciektr) [reply lib/versions.js:20]: We can just have `versionInput = toolVersion ?? "latest"` and move the if after this one for readability, but this is a super-nitpick

**Review by mkaput — COMMENTED** [review]
Looks good! Please add a test and LGTM 🎉

**Review by mkaput — COMMENTED** [review]
Looks good. Please rebase commits so that they’ll have a linear meaning and I’m happy to accept this :)

**Review by mkaput — APPROVED** [review]

**Pull request merged**